archunit/src/test/java/com/tngtech/archunit/core/domain/DependencyTest.java

```diff
@@ -71,6 +71,31 @@ public void Dependency_from_throws_declaration() {
                 .contains("Method <" + origin.getFullName() + "> throws type <" + IOException.class.getName() + ">");
     }
 
+    @Test
+    public void Dependency_from_instanceof_check_in_method() {
+        JavaMethod origin = importClassesWithContext(ClassWithDependencyOnInstanceofCheck.class, SomeMemberType.class).get(ClassWithDependencyOnInstanceofCheck.class)
+                .getMethod("method", Object.class);
+        InstanceofCheck instanceofCheck = getOnlyElement(origin.getInstanceofChecks());
+
+        Dependency dependency = Dependency.tryCreateFromInstanceofCheck(instanceofCheck).get();
+
+        assertThat(dependency.getOriginClass()).matches(ClassWithDependencyOnInstanceofCheck.class);
+        assertThat(dependency.getTargetClass()).matches(SomeMemberType.class);
+        assertThat(dependency.getDescription()).as("description")
+                .contains("Method <" + origin.getFullName() + "> checks instanceof <" + SomeMemberType.class.getName() + ">" + " in (DependencyTest.java:2"); // do not check exact location as line will move if file changes
+        assertThat(dependency.getSourceCodeLocation()).isEqualTo(origin.getSourceCodeLocation());
+    }
+
+    @Test
+    public void Dependency_from_instanceof_check_in_static_initializer() {
+        JavaClasses javaClasses = importClassesWithContext(ClassWithDependencyOnInstanceofCheck.class, SomeMemberType.class);
```

Comment:
Shouldn't this test actually use `ClassWithDependencyInStaticInitializerOnInstanceofCheck`? Somehow that gives me the feeling that this test was not written TDD :stuck_out_tongue_closed_eyes: Also why has the test a different format than the test before? Shouldn't the assertions be uniform? (see `@DataProvider`)

Reply:
integrated this test in the previous one 😃